lib/rubocop/cop/vendor/disallow_after_commit.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Vendor
+      class DisallowAfterCommit < Base
+        extend AutoCorrector
+
+        NO_METH_MSG = "You should not use `%<old>s` %<desc>s"
+        NO_METHS = {
+          after_commit: "generally you want `before_commit` as after_commit is non-transactional in guarentees of running",
```

Comment:
Suggestion for clarity and using imperative language. 
```suggestion
          after_commit: "use `before_commit` because after_commit is non-transactional an may not run",
```